pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
```

Comment:
What happens when we bust up controller-manager into other binaries?  Is this a type in the sense that it can have multiple instances or is it a unique identifier?